pkg/provider/provider.go

```diff
@@ -218,278 +158,184 @@ func (p *Provider) getVaultURL(ctx context.Context) (vaultURL *string, err error
 		}
 	}
-	if usePodIdentity {
-		log.Infof("using pod identity to access keyvault")
-		if p.PodName == "" || p.PodNamespace == "" {
-			return fmt.Errorf("pod information is not available. deploy a CSIDriver object to set podInfoOnMount")
+	if useVMManagedIdentityStr != "" {
+		useVMManagedIdentity, err = strconv.ParseBool(useVMManagedIdentityStr)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse useVMManagedIdentity flag, error: %+v", err)
 		}
-		log.Infof("mounting secrets store object content for %s/%s", p.PodNamespace, p.PodName)
-	} else if useVMManagedIdentity {
-		log.Infof("using vm managed identity to access keyvault")
 	}
-	if useVMManagedIdentity {
-		log.Infof("using vmss user identity to access keyvault")
+
+	p.AuthConfig, err = auth.NewConfig(usePodIdentity, useVMManagedIdentity, userAssignedIdentityID, secrets)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create auth config, error: %+v", err)
 	}
 
 	objectsStrings := attrib["objects"]
 	if objectsStrings == "" {
-		return fmt.Errorf("objects is not set")
+		return nil, fmt.Errorf("objects is not set")
 	}
-	log.Infof("objects: %s", objectsStrings)
+	klog.V(2).Infof("objects: %s", objectsStrings)
 
 	var objects StringArray
 	err = yaml.Unmarshal([]byte(objectsStrings), &objects)
 	if err != nil {
-		log.Infof("unmarshal failed for objects")
-		return err
+		return nil, fmt.Errorf("failed to yaml unmarshal objects, error: %+v", err)
 	}
-	log.Debugf("objects array: %v", objects.Array)
-	keyVaultObjects := []KeyVaultObject{}
+	klog.V(2).Infof("objects array: %v", objects.Array)
+	var keyVaultObjects []KeyVaultObject
 	for i, object := range objects.Array {
 		var keyVaultObject KeyVaultObject
 		err = yaml.Unmarshal([]byte(object), &keyVaultObject)
 		if err != nil {
-			log.Infof("unmarshal failed for keyVaultObjects at index %d", i)
-			return err
+			return nil, fmt.Errorf("unmarshal failed for keyVaultObjects at index %d", i)
 		}
 		keyVaultObjects = append(keyVaultObjects, keyVaultObject)
 	}
 
-	log.Infof("unmarshaled keyVaultObjects: %v", keyVaultObjects)
-	log.Infof("keyVaultObjects len: %d", len(keyVaultObjects))
+	klog.Infof("unmarshaled keyVaultObjects: %v", keyVaultObjects)
+	klog.Infof("keyVaultObjects len: %d", len(keyVaultObjects))
 
 	if len(keyVaultObjects) == 0 {
-		return fmt.Errorf("objects array is empty")
+		return nil, fmt.Errorf("objects array is empty")
 	}
 	p.KeyvaultName = keyvaultName
 	p.AzureCloudEnvironment = azureCloudEnv
-	p.UsePodIdentity = usePodIdentity
-	p.UseVMManagedIdentity = useVMManagedIdentity
-	p.UserAssignedIdentityID = userAssignedIdentityID
 	p.TenantID = tenantID
 
+	objectVersionMap := make(map[string]string)
 	for _, keyVaultObject := range keyVaultObjects {
-		log.Infof("fetching object: %s, type: %s from key vault", keyVaultObject.ObjectName, keyVaultObject.ObjectType)
+		klog.Infof("fetching object: %s, type: %s from key vault %s", keyVaultObject.ObjectName, keyVaultObject.ObjectType, p.KeyvaultName)
```

Comment:
should this be `klog.V(2).Infof` to be consistent with when we want to log these values?